// @rblmdst/scheval/dist/cjs/validators/string.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isEmail = exports.isEnumOf = exports.hasMatch = exports.hasMaxLength = exports.hasMinLength = exports.isString = void 0;
var emailPattern = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
var isString = function (v) {
    return typeof v === "string";
};
exports.isString = isString;
var hasMinLength = function (v, minLength, trim) {
    if (trim === void 0) { trim = false; }
    return trim ? v.trim().length >= minLength : v.length >= minLength;
};
exports.hasMinLength = hasMinLength;
var hasMaxLength = function (v, maxLength, trim) {
    if (trim === void 0) { trim = false; }
    return trim ? v.trim().length <= maxLength : v.length <= maxLength;
};
exports.hasMaxLength = hasMaxLength;
var hasMatch = function (v, pattern) {
    return v.match(pattern);
};
exports.hasMatch = hasMatch;
var isEnumOf = function (v, enums) {
    return enums.indexOf(v) !== -1;
};
exports.isEnumOf = isEnumOf;
var isEmail = function (v) {
    return v.match(emailPattern);
};
exports.isEmail = isEmail;
