// @rblmdst/scheval/dist/cjs/validators-map.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validatorsMap = void 0;
var validators_1 = require("./validators");
var array_1 = require("./validators/array");
var number_1 = require("./validators/number");
exports.validatorsMap = {
    string: {
        optional: validators_1.isOptional,
        required: validators_1.isDefined,
        type: validators_1.isString,
        min: validators_1.hasMinLength,
        max: validators_1.hasMaxLength,
        match: validators_1.hasMatch,
        enum: validators_1.isEnumOf,
        email: validators_1.isEmail,
    },
    boolean: { type: validators_1.isBoolean, optional: validators_1.isOptional, required: validators_1.isDefined },
    number: {
        optional: validators_1.isOptional,
        required: validators_1.isDefined,
        type: validators_1.isNumber,
        integer: validators_1.isInteger,
        le: validators_1.isLowerOrEqualTo,
        ge: validators_1.isGreaterOrEqualTo,
        eq: validators_1.isEqualTo,
        gt: validators_1.isGreaterThan,
        lt: validators_1.isLowerThan,
        btw: number_1.isInRange,
    },
    array: {
        type: validators_1.isArray,
        optional: validators_1.isOptional,
        required: validators_1.isDefined,
        ofType: validators_1.isArrayOfType,
        ofMinSize: validators_1.hasMinSize,
        ofMaxSize: validators_1.hasMaxSize,
        ofSize: array_1.hasSize,
    },
    object: {
        type: validators_1.isObject,
        optional: validators_1.isOptional,
        required: validators_1.isDefined,
    },
};
